### PR TITLE
Detect loop in Linked list JAVA

### PR DESCRIPTION
**Pull request by yashaswibiyahut**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

**YES**

### Description
Issue no: #686
**Approach**: Floyd’s Cycle-Finding Algorithm
 This is the fastest method to solve this problem.
_Time complexity_: O(n).
_Auxiliary Space_:O(1).

### Checklist

- [x] I've read the contribution guidelines.
- [x]  I've checked the issue list before deciding what to submit.
- [x]  I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

(Write your answer here.)

### CODE REVIEW

**Comment by yashaswibiyahut** [comment]
kindly review this @Kajol-Kumari @SKAUL05 @siddharth25pandey @iamrajiv @HarshCasper

File: Java/ds/detect_loop.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+		// we reach here if list does not contain any cycle
+		return false;
+	}
+
+	// Detect Cycle in a linked list
+	public static void main(String[] args)
+	{
+		// input keys
+		int[] keys = {1, 2, 3, 4, 5};
```
Comment (siddharth25pandey) [inline Java/ds/detect_loop.java:47]: Take Dynamic input from the user by using **Scanner**

File: Java/ds/detect_loop.java [hunk 2]
```diff
@@ -0,0 +1,62 @@
+import java.util.HashSet;
```
Comment (siddharth25pandey) [inline Java/ds/detect_loop.java:1]: Above this line add a brief Description about the Program

File: Java/ds/detect_loop.java [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+		// insert cycle
+		head.next.next.next.next.next = head.next.next;
+
+		if (detectCycle(head)) {
+			System.out.println("Cycle Found");
+		} else {
+			System.out.println("No Cycle Found");
+		}
+	}
```
Comment (siddharth25pandey) [inline Java/ds/detect_loop.java:61]: Add Sample Input and Output as a Comment

**Review by siddharth25pandey — CHANGES_REQUESTED** [review]

**Comment by siddharth25pandey** [comment]
This code is exactly copied from [this](https://www.geeksforgeeks.org/detect-loop-in-a-linked-list/)
**Don't copy and plagiarize code directly from anywhere. In case of repeated violations a strict action will be taken & you will be prohibited to contribute in this repo.**
 # **Count 2**

**Pull request closed without merging**